dlt/common/storages/fsspec_filesystem.py

```diff
@@ -144,6 +144,8 @@ def fsspec_from_config(config: FilesystemConfiguration) -> Tuple[AbstractFileSys
     fs_kwargs = prepare_fsspec_args(config)
 
     try:
+        if config.bucket_url.startswith("file://"):
+            fs_kwargs["auto_mkdir"] = True
```

Comment:
@rudolfix is it a good place to enable `auto_mkdir`?

Reply:
So I figured this out after few trials we can pass `kwargs = { auto_mkdir = true }` for the cases when layout has deeper nesting.